### PR TITLE
Fix broken version check on metadata section

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: src/binary-reader.cc [hunk 1]
```diff
@@ -1459,7 +1459,7 @@ Result BinaryReader::ReadLinkingSection(Offset section_size) {
   CALLBACK(BeginLinkingSection, section_size);
   uint32_t version;
   CHECK_RESULT(ReadU32Leb128(&version, "version"));
-  ERROR_UNLESS(version = 1, "invalid linking metadata version");
+  ERROR_UNLESS(version == 1, "invalid linking metadata version: %u", version);
```
Comment (binji) [inline src/binary-reader.cc]: brutal!

**Comment by shepmaster** [comment]
Howdy! This exposes an error when using WASM generated by Rust:

```
error: invalid linking metadata version: 3
```

Is there a flag I should be using for this?

**Comment by shepmaster** [comment]
Perhaps related to the discussion in #838

**Comment by sbc100** [comment]
Yes, you may be able to use `--ignore-custom-section-errors`.  The problem should fix itself once Rust pulls  in a new llvm version.

**Comment by shepmaster** [comment]
Hmm, the flag doesn't seem to take effect; perhaps I'm doing something wrong?

```
$ wasm2wat --help | grep ignore-custom
      --ignore-custom-section-errors          Ignore errors in custom sections

$ wasm2wat --fold-exprs --generate-names --inline-exports --ignore-custom-section-errors -o /tmp/out.wat target/wasm32-unknown-unknown/debug/playground.wasm.gc
002f892: error: invalid linking metadata version: 3
```

**Comment by shepmaster** [comment]
Moved discussion to #842.

**Pull request merged**